File: 929.独特的电子邮件地址.py
#
# @lc app=leetcode.cn id=929 lang=python3
#
# [929] 独特的电子邮件地址
#

# @lc code=start
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        emailSet = set()
        for email in emails:
            index = email.index("@")
            local_name = email[:index].split("+", 1)[0]
            local_name = local_name.replace(".", "")
            emailSet.add(local_name + email[index:])
        return len(emailSet)
# @lc code=end

File: 929.独特的电子邮件地址.go
/*
 * @lc app=leetcode.cn id=929 lang=golang
 *
 * [929] 独特的电子邮件地址
 */

// @lc code=start
func numUniqueEmails(emails []string) int {
	eamilMap := map[string]struct{}{}
	for _, email := range emails {
		index := strings.IndexByte(email, '@')
		localName := strings.SplitN(email[:index], "+", 2)[0]
		localName = strings.ReplaceAll(localName, ".", "")
		eamilMap[localName+email[index:]] = struct{}{}
	}
	return len(eamilMap)
}

// @lc code=end

